request.go

```diff
@@ -27,7 +27,89 @@ import (
+	// from the source_file_descriptors field used for the files in file_to_geneate if WithSourceRetentionOptions
+	// is specified.
+	//
+	// The caller can assume that all FileDescriptors have a valid path as the name field.
+	// Paths are considered valid if they are non-empty, relative, use '/' as the path separator, do not jump context,
+	// and have `.proto` as the file extension.
+	AllFiles() (*protoregistry.Files, error)
+	// FileDescriptorProtosToGenerate returns the FileDescriptors for the files specified by the
+	// file_to_generate field.
+	//
+	// The caller can assume that all FileDescriptorProtoss have a valid path as the name field.
+	// Paths are considered valid if they are non-empty, relative, use '/' as the path separator, do not jump context,
+	// and have `.proto` as the file extension.
+	FileDescriptorProtosToGenerate() []*descriptorpb.FileDescriptorProto
+	// AllFileDescriptorProtos returns the FileDescriptorProtos for all files in the CodeGeneratorRequest.
+	//
+	// This matches with the proto_file field on the CodeGeneratorRequest, with the FileDescriptorProtos
+	// from the source_file_descriptors field used for the files in file_to_geneate if WithSourceRetentionOptions
+	// is specified.
+	//
+	// The caller can assume that all FileDescriptorProtoss have a valid path as the name field.
+	// Paths are considered valid if they are non-empty, relative, use '/' as the path separator, do not jump context,
+	// and have `.proto` as the file extension.
+	AllFileDescriptorProtos() []*descriptorpb.FileDescriptorProto
+	// CompilerVersion returns the specified compiler_version on the CodeGeneratorRequest.
+	//
+	// If the compiler_version field was not present, nil is returned.
+	//
+	// The caller can assume that the major, minor, and patch values are non-negative.
+	CompilerVersion() *CompilerVersion
+	// CodeGeneratorRequest returns the raw underlying CodeGeneratorRequest.
+	//
+	// The returned CodeGeneratorRequest is a not copy - do not modify it! If you would
+	// like to modify the CodeGeneratorRequest, use proto.Clone to create a copy.
+	CodeGeneratorRequest() *pluginpb.CodeGeneratorRequest
+	// WithSourceRetentionOptions will return a copy of the Request that will result in all
+	// methods returning descriptors with source-retention options retained on files to generate.
+	//
+	// By default, only runtime-retention options are included on files to generate. Note that
+	// source-retention options are always included on files not in file_to_generate.
+	//
+	// An error will be returned if the underlying CodeGeneratorRequest did not have source_file_descriptors populated.
+	WithSourceRetentionOptions() (Request, error)
+
+	isRequest()
+}
+
+// NewRequest returns a new Request for the CodeGeneratorRequest.
+//
+// The CodeGeneratorRequest will be validated as part of construction.
+func NewRequest(codeGeneratorRequest *pluginpb.CodeGeneratorRequest) (Request, error) {
```

Comment:
Okay, this is making more sense now why it should be in the same package. Disregard my suggestion to move this out into a sub-package.